precompiles/xcm-utils/src/lib.rs

```diff
@@ -73,4 +79,78 @@ where
 			.into();
 		Ok(Address(account))
 	}
+
+	#[precompile::public("getUnitsPerSecond((uint8,bytes[]))")]
+	#[precompile::view]
+	fn get_units_per_second(
+		handle: &mut impl PrecompileHandle,
+		multilocation: MultiLocation,
+	) -> EvmResult<u128> {
+		// TODO: Change once precompiles are benchmarked
+		// for now we charge a db read,
+		handle.record_cost(RuntimeHelper::<Runtime>::db_read_gas_cost())?;
+
+		// We will construct an asset with the max amount, and check how much we
+		// get in return to substract
+		let multiasset: xcm::latest::MultiAsset = (multilocation.clone(), u128::MAX).into();
+		let payment: xcm_executor::Assets = vec![multiasset].into();
+		let weight_per_second = 1_000_000_000_000u64;
+		let mut trader = XcmConfig::Trader::new();
+		let remaining: Vec<xcm::latest::MultiAsset> = trader
+			.buy_weight(weight_per_second, payment.clone())
+			.map_err(|_| revert("error retrieving info"))?
+			.into();
+
+		// If remaining is empty, it means we spent the whole max u128,
+		// shouldnt happen
+		let remaining_asset = remaining
+			.first()
+			.ok_or(revert("spent whole weight, shouldnt happen"))?;
+
+		let paid_assets: Vec<xcm::latest::MultiAsset> = payment
+			.clone()
+			.checked_sub(remaining_asset.clone())
+			.map_err(|_| revert("error retrieving info"))?
```

Comment:
Better revert message ?

Reply:
fixed in fc4e9d8258c9f8c388cb12b2e06fb6cbcb97157f